src/main/java/com/puppycrawl/tools/checkstyle/AstTreeStringPrinter.java

```diff
@@ -114,10 +117,13 @@ private static String printJavaAndJavadocTree(DetailAST ast) {
      * @return string javadoc tree
      */
     private static String parseAndPrintJavadocTree(DetailAST node) {
-        final DetailNode tree = DetailNodeTreeStringPrinter.parseJavadocAsDetailNode(node);
+        final DetailAST javadocBlock = node.getParent();
+        final DetailNode tree = DetailNodeTreeStringPrinter.parseJavadocAsDetailNode(javadocBlock);
 
-        final String rootPrefix = getIndentation(node);
-        final String prefix = rootPrefix.substring(0, rootPrefix.length() - 2) + "   ";
+        String baseIdentation = getIndentation(node);
+        baseIdentation = baseIdentation.substring(0, baseIdentation.length() - 2);
+        final String rootPrefix = baseIdentation + "   `--";
+        final String prefix = baseIdentation + "       ";
```

Comment:
@romani , ``getIndentation()`` processes only DetailAST nodes and it cannot calculate indentation for javadoc tree nodes because Main Tree (DetailAST) and Javadoc tree (DetailNode) are two different trees with no references between them. That's why we calculate prefixes for inner javadoc tree manualy (there was also a hack before: ``final String prefix = rootPrefix.substring(0, rootPrefix.length() - 2) + "   ";``).
However, I also don't like these hardcoded prefixes :)  I'm thinking how to change this...